util/cmake/SeqAnBuildSystem.cmake

```diff
@@ -47,30 +47,40 @@
 include (SeqAnUsabilityAnalyzer)
 include (CheckCXXCompilerFlag)
 
+set (COMPILER_CLANG FALSE)
+set (COMPILER_GNU FALSE)
+set (COMPILER_INTEL FALSE)
+set (COMPILER_WINTEL FALSE)
+set (COMPILER_MSVC FALSE)
+set (STDLIB_VS ${MSVC})
+
+if (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  set (COMPILER_CLANG TRUE)
+elseif (CMAKE_CXX_COMPILER_ID MATCHES "Intel" AND STDLIB_VS)
+  set (COMPILER_WINTEL TRUE)
+elseif (CMAKE_CXX_COMPILER_ID MATCHES "Intel")
+  set (COMPILER_INTEL TRUE)
+elseif (CMAKE_CXX_COMPILER_ID MATCHES "GNU")
+  set (COMPILER_GNU TRUE)
+elseif (CMAKE_CXX_COMPILER_ID MATCHES "MSVC")
+  set (COMPILER_MSVC TRUE)
+endif ()
+
 # ---------------------------------------------------------------------------
 # Enable /bigobj flag on Windows.
 # ---------------------------------------------------------------------------
 
 # We need the /bigobj switch on windows (for 64 bit builds only actually).
 # Set target system to be Windows Vista and later.
-if (MSVC)
-  add_definitions (/bigobj /D_WIN32_WINNT=0x0600 /DWINVER=0x0600)
-elseif (MINGW)
-  add_definitions (-D_WIN32_WINNT=0x0600 -DWINVER=0x0600)
-endif (MSVC)
-
-# ---------------------------------------------------------------------------
-# Set architecture for MinGW.
-#
-# If we do not set i586 as the architecture for MinGW then generating atomic
-# expressions will fail.
-# ---------------------------------------------------------------------------
+if (COMPILER_MSVC OR COMPILER_WINTEL)
+    # Set /bigobj for COMPILER_MSVC and COMPILER_WINTEL, but COMPILER_CLANG on
+    # windows (clang/c2 3.7) can not handle it.
+    add_definitions (/bigobj)
```

Comment:
It does work now on clang/c2 and we should set it:
https://blogs.msdn.microsoft.com/vcblog/2016/03/31/clang-with-microsoft-codegen-march-2016-released/